### PR TITLE
Fix Bug for OpenAPI 3 Emitter crash on @useAuth({}) 

### PR DESCRIPTION
**Pull request by skywing918**
OpenAPI 3 Emitter crash on @useAuth({}). 
seems the diagnostics is not initial before add item. 

fix: https://github.com/microsoft/typespec/issues/4097

### CODE REVIEW

File: packages/openapi3/src/openapi.ts [hunk 1]
```diff
@@ -306,6 +306,7 @@ function createOAPIEmitter(
       } as any,
       context
     );
+    diagnostics = createDiagnosticCollector();
```
Comment (wanlwanl) [inline packages/openapi3/src/openapi.ts:309]: can we move it to line 213?

Comment (timotheeguerin) [reply packages/openapi3/src/openapi.ts:309]: no this needs to be reset for each service, but could move line 289 I think

**Review by markcowl — CHANGES_REQUESTED** [review]
Note: you will need to add a changelog entry for the PR to be merged.

Also, if possible, should add a test to ensure no emitter crash.

File: packages/openapi3/test/security.test.ts [hunk 2]
```diff
@@ -246,6 +247,21 @@ describe("openapi3: security", () => {
...
+      @service
+      @useAuth({})
+      namespace MyService {}
+      `
+    );
+    expectDiagnostics(diagnostics, [
+      {
+        code: "@typespec/openapi3/unsupported-auth",
+      },
+    ]);
```
Comment (timotheeguerin) [inline packages/openapi3/test/security.test.ts:262]:
```suggestion
    expectDiagnostics(diagnostics, {
      code: "@typespec/openapi3/unsupported-auth",
    });
```

don't need the extra array when testing a single diagnostic

**Review by chrisradek — APPROVED** [review]
Changes look good to me - also tested and confirmed they fix #4272 as well (empty model vs empty model expression)

**Review by markcowl — APPROVED** [review]
